src/main/java/lotto/domain/Lotto.java

```diff
@@ -7,11 +7,17 @@
 
     private static final int LOTTO_SIZE = 6;
 
-    private final Set<LottoNumber> lottoNumbers;
+    private final List<LottoNumber> lottoNumbers;
 
     private Lotto(Set<LottoNumber> lottoNumbers) {
         checkLottoSize(lottoNumbers);
-        this.lottoNumbers = Collections.unmodifiableSet(lottoNumbers);
+        this.lottoNumbers = Collections.unmodifiableList(sortedList(lottoNumbers));
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
        this.lottoNumbers = Collections.unmodifiableSet(new TreeSet<>(lottoNumbers));
```